Sources/TuistDependencies/DependenciesController.swift

```diff
@@ -191,20 +191,28 @@ public final class DependenciesController: DependenciesControlling {
 
 extension TuistCore.DependenciesGraph {
     public func merging(with other: Self) throws -> Self {
-        let mergedExternalDependencies = try other.externalDependencies.reduce(into: externalDependencies) { result, entry in
-            if let alreadyPresent = result[entry.key] {
-                throw DependenciesControllerError.duplicatedDependency(entry.key, alreadyPresent, entry.value)
-            }
 
-            result[entry.key] = entry.value
+        // Per default set it to the own externalDependencies, avoid missing dependencies for a platform that is missing in `other`
+        var mergedExternalDependencies: [ProjectDescription.Platform: [String: [ProjectDescription.TargetDependency]]] = externalDependencies
+
+        // Then iterate over all platforms of `other` and merge them into self.externalDependencies for the specified platform
+        try other.externalDependencies.forEach { platform, otherPlatformDependencies in
+            let mergedPlatformDependencies = try otherPlatformDependencies.reduce(into: externalDependencies[platform] ?? [:]) { result, entry in
+                if let alreadyPresent = result[entry.key] {
+                    throw DependenciesControllerError.duplicatedDependency(entry.key, alreadyPresent, entry.value)
+                }
+                result[entry.key] = entry.value
+            }
+            mergedExternalDependencies[platform] = mergedPlatformDependencies
         }
```

Comment:
As above, given we already have the dictionary is simpler to use a forEach rather than a reduce. As it's simpler, you probably don't even need the comments
```suggestion
            var mergedExternalDependencies = externalDependencies
            try otherPlatformDependencies.forEach { name, dependency in
                if let alreadyPresent = mergedExternalDependencies[platform]?[name] {
                    throw DependenciesControllerError.duplicatedDependency(name, alreadyPresent, dependency)
                }
                mergedExternalDependencies[platform, default: [:]][name] = dependency
            }
```